ruby/ql/lib/codeql/ruby/frameworks/ActionController.qll

```diff
@@ -35,16 +22,13 @@ private class ApplicationControllerAccess extends ConstantReadAccess {
  */
 class ActionControllerControllerClass extends ClassDeclaration {
   ActionControllerControllerClass() {
-    // class FooController < ActionController::Base
-    this.getSuperclassExpr() instanceof ActionControllerBaseAccess
-    or
-    // class FooController < ApplicationController
-    this.getSuperclassExpr() instanceof ApplicationControllerAccess
-    or
-    // class BarController < FooController
-    exists(ActionControllerControllerClass other |
-      other.getModule() = resolveConstantReadAccess(this.getSuperclassExpr())
-    )
+    this.getSuperclassExpr() =
+      [
+        API::getTopLevelMember("ActionController").getMember("Base"),
+        // In Rails applications `ApplicationController` typically extends `ActionController::Base`, but we
+        // treat it separately in case the `ApplicationController` definition is not in the database.
+        API::getTopLevelMember("ApplicationController")
+      ].getASubclass*().getAUse().asExpr().getExpr()
```

Comment:
I think it would be nice to have a method named `getADirectSubclass` and define `getASubclass` to be `getADirectSubclass*`. I expect users in almost all cases need the transitive closure and they are likely to forget adding `*`. Having the nicely named method "just do the right thing in most cases" would be helpful.